### PR TITLE
Add Traffic Boost settings infrastructure

### PR DESCRIPTION
**Pull request by acicovic**
## Description
This PR introduces a Traffic Boost settings infrastructure (to be populated in the future with real settings), passing the setting values into our new dashboard page along with the current user's PCH permissions.

## Motivation and context
Being able to pass settings and permissions into the dashboard page to tailor the experience accordingly.

## How has this been tested?
Manual local testing.

## Screenshots
image

## Summary by CodeRabbit

- **New Features**
	- Introduced dynamic permissions and settings for Content Helper and Traffic Boost in the dashboard.
	- Added a new API endpoint for managing Traffic Boost settings.
	- Enhanced the dashboard page to display dynamic content based on retrieved settings and permissions.
	- Expanded settings structure to include Traffic Boost options.

- **Bug Fixes**
	- Improved error handling for settings retrieval and validation.

- **Documentation**
	- Updated comments and documentation for new classes and methods.

### CODE REVIEW

File: src/content-helper/dashboard-page/pages/dashboard/page-component.tsx [hunk 1]
```diff
@@ -1,13 +1,72 @@
...
+	// Merge parsed settings with default settings.
+	const mergedSettings = { ...defaultSettings, ...parsedSettings };
+
+	// Fix invalid values if any are found.
+	if ( typeof mergedSettings.Setting1 !== 'string' ) {
+		mergedSettings.Setting1 = defaultSettings.Setting1;
+	}
+
+	return parsedSettings;
+};
```
[the remark this answers is not in the document]
Comment (acicovic) [reply src/content-helper/dashboard-page/pages/dashboard/page-component.tsx]: Good catch. Should be fixed now.

File: src/content-helper/dashboard-page/pages/dashboard/page-component.tsx [hunk 2]
```diff
@@ -1,13 +1,72 @@
...
+			endpoint="traffic-boost"
+			defaultSettings={ getSettingsFromJson( window.wpParselyContentHelperSettings ) }
+		>
+			<VerifyCredentials>
+				<>
+					<h1>Parse.ly</h1>
+					<p>Welcome to the Parse.ly Dashboard page!</p>
+					<p>Content Helper Permissions: { JSON.stringify( getContentHelperPermissions() ) }</p>
+					<p>Traffic Boost Settings: { JSON.stringify( getSettingsFromJson( window.wpParselyContentHelperSettings ) ) }</p>
+				</>
```
[the remark this answers is not in the document]
Comment (acicovic) [reply src/content-helper/dashboard-page/pages/dashboard/page-component.tsx:68]: This is temporary code that will be reworked.

File: src/UI/class-dashboard-page.php [hunk 3]
```diff
@@ -168,11 +157,62 @@ public function enqueue_dashboard_page_scripts( ?string $hook_suffix ): void {
...
+			$settings = array();
+		}
+
+		$settings = wp_json_encode( $settings );
+
+		wp_add_inline_script(
+			'parsely-dashboard-page',
+			"window.wpParselyContentHelperSettings = '$settings';",
+			'before'
+		);
```
[the remark this answers is not in the document]
Comment (acicovic) [reply src/UI/class-dashboard-page.php:216]: This variable is passed through `wp_json_encode` on line 210.

File: src/UI/class-dashboard-page.php [hunk 4]
```diff
@@ -168,11 +157,62 @@
...
+		$permissions_json = Permissions::get_pch_permissions_json(
+			$this->parsely->get_options()['content_helper']
+		);
+
+		wp_add_inline_script(
+			'parsely-dashboard-page',
+			"window.wpParselyContentHelperPermissions = '$permissions_json';",
+			'before'
+		);
+	}
```
[the remark this answers is not in the document]
Comment (acicovic) [reply src/UI/class-dashboard-page.php:187]: The escaping of the data is ensured by the `get_pch_permissions_json()` function which uses `wp_json_encode()`

**Comment by vaurdan** [comment]
The code looks good to me! I do have a more conceptual question though: instead of calling it `traffic-boost-settings`, should we be calling it `dashboard-settings`? 

The way I see it, Traffic Boost is a feature that lives inside the Parse.ly Dashboard (or whatever we're calling the SPA we're building, I'm still not sure Dashboard is the best name, since effectively the Dashboard will only be the first page). Comparing, for example, with the Editor Sidebar, which has its own settings namespace, and the individual features are part of that namespace - title suggestions, smart linking, etc. 

I think that making the settings namespace more broad, instead of pinning it to the Traffic Boost, will allow for easier expansion in the future, if/when we add more features to the so called "Dashboard".

**Comment by acicovic** [comment]
@vaurdan, this is a fair point.

The difficulty of the moment is that we don't know how our first iteration of this will end up. We might very well finish with a single page that only needs TB settings.

If the only settings we're having are TB related, then leaving it as is would be fine. If we're working with a broader range of settings, then I'd go with the general name.

I think that sooner or later we'll get to what you suggest though.

As things stand, for me both options are fine. Do you think it's better to go with the broader name right away? I think we'll need some more code for that.

**Comment by vaurdan** [comment]
I do think we might end up with settings on the dashboard main page sooner than later - filtering options, for example - so IMO it makes sense to make it broader now and save the trouble later.

**Review by vaurdan — APPROVED** [review]

**Pull request merged**